src/Converter.cc

```diff
@@ -229,13 +240,334 @@ void Converter::ConvertImpl(tinyxml2::XMLElement *_elem,
+                                                const size_t &_newNameIdx)
+{
+  Converter::TupleVector deleteElems;
+  std::string newModelName = _newModel->Attribute("name");
+
+  tinyxml2::XMLElement *elem = _copy->FirstChildElement(), *nextElem = nullptr;
+  while (elem)
+  {
+    nextElem = elem->NextSiblingElement();
+
+    std::string elemName = elem->Name();
+    std::string elemAttrName;
+
+    if (elem->Attribute("name"))
+      elemAttrName = elem->Attribute("name");
+
+    if (elemAttrName.empty() ||
+        elemAttrName.compare(0, _newNameIdx - 2, newModelName) != 0 ||
+        (elemName != "frame" && elemName != "joint" &&
+        elemName != "link" && elemName != "model"))
+    {
+      if (elemName != "include" && elemName != "gripper")
+      {
+        elem = nextElem;
+        continue;
+      }
+    }
+
+    // Child attribute name w/ newModelName prefix stripped
+    // except for //include & //gripper
+    std::string childAttrName;
+    if (elemName != "include" && elemName != "gripper")
+    {
+      childAttrName = elemAttrName.substr(_newNameIdx);
+      elem->SetAttribute("name", childAttrName.c_str());
+    }
+
+    // strip new model prefix from //pose/@relative_to
+    tinyxml2::XMLElement *poseElem = elem->FirstChildElement("pose");
+    if (poseElem != nullptr && poseElem->Attribute("relative_to"))
+    {
+      std::string poseRelTo = poseElem->Attribute("relative_to");
+
+      // strip new model prefix from //pose/@relative_to
+      if (poseRelTo.compare(0, _newNameIdx - 2, newModelName) == 0)
+      {
+        poseRelTo = poseRelTo.substr(_newNameIdx);
+        poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+      }
+      else if (elemName == "include")
+      {
+        // if //include/pose/@relative_to does not have new model prefix
+        // don't add to new model
+        elem = nextElem;
+        continue;
+      }
+    }
+
+    if (elemName == "frame")
+    {
+      std::string attachedTo;
+
+      if (elem->Attribute("attached_to"))
+      {
+        attachedTo = elem->Attribute("attached_to");
+
+        SDF_ASSERT(attachedTo.compare(0, _newNameIdx - 2, newModelName) == 0,
+          "Error: Frame attribute 'attached_to' does not start with " +
+          newModelName);
+
+        // strip new model prefix from attached_to
+        attachedTo = attachedTo.substr(_newNameIdx);
+        elem->SetAttribute("attached_to", attachedTo.c_str());
+      }
+
+      // remove frame if childAttrName == __model__
+      if (childAttrName == "__model__" && elem->Attribute("attached_to"))
+      {
+        _newModel->SetAttribute("canonical_link", attachedTo.c_str());
+        _newModel->InsertFirstChild(poseElem);
+
+        _copy->DeleteChild(elem);
+        elem = poseElem;
+      }
+    }  // frame
+
+    else if (elemName == "link")
+    {
+      // find & strip new model prefix of all //link/<element>/pose/@relative_to
+      for (tinyxml2::XMLElement *e = elem->FirstChildElement();
```

Comment:
One more: `//sensor/camera/pose` here and in the "elemName == joint" branch

Reply:
dce6828593004010ad58c44ab45f8d0b7bf462be